### PR TITLE
Threadify file logger, handle save files form cmdline

### LINKED ISSUE #638: EXEC_CRASH on invite from groupbot when already in a split groupchat
So... this happened http://tox.lighthouseapp.com/projects/123754-tox/tickets/10-qtox-crash-on-invite

### PR DESCRIPTION
**Pull request by dubslow**
fixes #638

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -39,7 +42,8 @@ void myMessageHandler(QtMsgType type, const QMessageLogContext& ctxt, const QStr
             && msg == QString("QFSFileEngine::open: No file name specified"))
         return;
 
-    dflt(type, ctxt, msg);
+    dflt(type, ctxt, msg); // this must be thread safe, otherwise qDebug() would never ever work
+    QMutexLocker locker(&mutex);
```
Comment (tux3) [inline src/main.cpp:45]: It probably works, but if we're going to bother locking anyway, why not guard the entire function just to be safe :) ?

File: qtox.pro [hunk 2]
```diff
@@ -214,4 +215,5 @@ SOURCES += \
     src/misc/flowlayout.cpp \
     src/widget/toxuri.cpp \
     src/toxdns.cpp \
-    src/ipc.cpp
+    src/ipc.cpp \
+    src/widget/toxsave.cpp
```
Comment (tux3) [inline qtox.pro:219]: You didn't include those in the PR (?)

**Comment by tux3** [comment]
Thanks for the copyright headers, I always forget those!

File: qtox.pro (hunk 2 above)
Comment (dubslow) [inline qtox.pro:219]: Every time, I swear...

**Pull request merged**